packages/catalog-process/test/db.test.ts

```diff
@@ -68,14 +117,1079 @@ describe("database test", async () => {
+          producerId: organizationId1,
+        };
+        await addOneEService(eService2);
+
+        const descriptor3: Descriptor = {
+          ...mockDescriptor,
+          id: uuidv4(),
+          state: descriptorState.published,
+        };
+        const eService3: EService = {
+          ...mockEService,
+          id: uuidv4(),
+          name: "eService 003",
+          descriptors: [descriptor3],
+          producerId: organizationId1,
+        };
+        await addOneEService(eService3);
+
+        const descriptor4: Descriptor = {
+          ...mockDescriptor,
+          id: uuidv4(),
+          state: descriptorState.draft,
+        };
+        const eService4: EService = {
+          ...mockEService,
+          id: uuidv4(),
+          name: "eService 004",
+          producerId: organizationId2,
+          descriptors: [descriptor4],
+        };
+        await addOneEService(eService4);
+
+        const descriptor5: Descriptor = {
+          ...mockDescriptor,
+          id: uuidv4(),
+          state: descriptorState.published,
+        };
+        const eService5: EService = {
+          ...mockEService,
+          id: uuidv4(),
+          name: "eService 005",
+          producerId: organizationId2,
+          descriptors: [descriptor5],
+        };
+        await addOneEService(eService5);
+
+        const descriptor6: Descriptor = {
+          ...mockDescriptor,
+          id: uuidv4(),
+          state: descriptorState.draft,
+        };
+        const eService6: EService = {
+          ...mockEService,
+          id: uuidv4(),
+          name: "eService 006",
+          producerId: organizationId2,
+          descriptors: [descriptor6],
+        };
+        await addOneEService(eService6);
+
+        const result1 = await readModelService.getEServices(
+          getMockAuthData(),
+          {
+            eservicesIds: [eService1.id, eService2.id],
+            producersIds: [],
+            states: [],
+            agreementStates: [],
+          },
+          0,
+          50
+        );
+        const result2 = await readModelService.getEServices(
+          getMockAuthData(),
+          {
+            eservicesIds: [],
+            producersIds: [organizationId1],
+            states: [],
+            agreementStates: [],
+          },
+          0,
+          50
+        );
+        const result3 = await readModelService.getEServices(
+          getMockAuthData(),
+          {
+            eservicesIds: [],
+            producersIds: [],
+            states: ["Draft"],
+            agreementStates: [],
+          },
+          0,
+          50
+        );
+        const result4 = await readModelService.getEServices(
+          getMockAuthData(),
+          {
+            eservicesIds: [],
+            producersIds: [organizationId2],
+            states: ["Draft"],
+            agreementStates: [],
+          },
+          0,
+          50
+        );
+        // TO DO test with other parameters configuration
```

Comment:
should we do this before closing the PR?

Reply:
We have to decide how many different combinations we want to test. Since this PR is already so huge, I thought of keeping this for a following PR.

Reply:
ok, same considerations about not forgetting this part

Reply:
Sure